### PR TITLE
IdentityX Omeda product subscription support

### PR DESCRIPTION
**Pull request by solocommand**
- [x] Requires parameter1/omeda#28

Adds support for sending product subscriptions for IdentityX questions with the `product` entity type.

### CODE REVIEW

**Comment by solocommand** [comment]
[Support `product` entity type subscription fields](https://trello.com/c/8CAohakI/21-support-product-entity-type-subscription-fields)

File: packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js [hunk 1]
```diff
@@ -63,15 +66,24 @@ module.exports = async ({
...
       omedaCustomer,
       fields: omedaLinkedFields.deploymentType,
     });
+    await setOmedaProducts({
+      identityX,
+      brandKey,
+      user,
+      omedaCustomer,
+      fields: omedaLinkedFields.product,
+    });
```
Comment (zarathustra323) [inline packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js]: Products and the `brandKey` option need to be handled in https://github.com/parameter1/base-cms/blob/master/packages/marko-web-omeda-identity-x/middleware/resync-customer-data.js

File: packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js [hunk 2]
```diff
@@ -63,15 +66,24 @@ module.exports = async ({
...
       fields: omedaLinkedFields.demographic,
     });
     await setOmedaDeploymentTypes({
       identityX,
+      brandKey,
       user,
       omedaCustomer,
       fields: omedaLinkedFields.deploymentType,
     });
+    await setOmedaProducts({
```
Comment (zarathustra323) [inline packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js]: Only sync each type if data is missing for that type. Do _not_ promise all these

File: packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js [hunk 3]
```diff
@@ -63,15 +66,24 @@ module.exports = async ({
   if (!hasAnsweredAllOmedaQuestions) {
     await setOmedaDemographics({
       identityX,
+      brandKey,
```
Comment (zarathustra323) [inline packages/marko-web-omeda-identity-x/integration-hooks/on-login-link-sent.js]: Move the sync operations to their own file and re-use in resync-customer-data... then add options for which types should actually fire....

```js
module.exports = async ({
  identityX,
  brandKey,
  user,
  omedaCustomer,
  omedaLinkedFields,
}, {
  shouldUpdateDemographics =  false,
  shouldUpdateDeploymentTypes =  false,
  shouldUpdateProducts =  false,
} = {}) => {
  if (shouldUpdateDemographics) {
    await setOmedaDemographics({
      identityX,
      brandKey,
      user,
      omedaCustomer,
      fields: omedaLinkedFields.demographic,
    });
  }
  // ...
}
```

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

File: packages/marko-web-omeda-identity-x/omeda-data/update-identity-x.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    omedaLinkedFields = {},
+  } = {},
+  {
+    updateData = false,
+    updateDemographics = false,
+    updateDeploymentTypes = false,
+    updateProducts = false,
+  } = {},
+) => {
+  console.log('Updating IdentityX Data', {
```
Comment (zarathustra323) [inline packages/marko-web-omeda-identity-x/omeda-data/update-identity-x.js]: Remove console.log

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**